File: pb/src/main/java/org/sqlbuilder/pb/objects/Predicate.java
package org.sqlbuilder.pb.objects;

public class Predicate extends LexItem
{
	private final String head;

	public Predicate(final String head, final String lemma)
	{
		super(lemma);
		this.head = head;
	}

	public static Predicate make(final String head, final String lemmaAttribute)
	{
		return new Predicate(head, lemmaAttribute);
	}

	public String getHead()
	{
		return this.head;
	}
}

File: pb/src/main/java/org/sqlbuilder/pb/objects/Predicate.kt
package org.sqlbuilder.pb.objects;

public class Predicate extends LexItem
{
	private final String head;

	public static Predicate make(final String head, final String lemma)
	{
		//assert head.equals(lemma) : head + "!=" + lemma;
		return new Predicate(head, lemma);
	}

	private Predicate(final String head, final String lemma)
	{
		super(lemma);
		this.head = head;
	}

	public String getHead()
	{
		return this.head;
	}
}
